### PR TITLE
feat(utils): add toPathname, hasSamePathname, allHaveSamePathname URL helpers

### PR DESCRIPTION
**Pull request by dhavkuma**
## Summary

- Adds `toPathname(url)` — extracts and normalizes a URL's pathname (strips trailing slashes on non-root paths, lowercases, falls back to the raw string for unparseable input)
- Adds `hasSamePathname(url, referenceUrl)` — case-insensitive, trailing-slash-tolerant pathname equality check between two URLs
- Adds `allHaveSamePathname(urls, referenceUrl)` — checks that every URL in an array shares the same normalized pathname as a reference

All three functions include JSDoc comments and are fully tested (15 new test cases, 100% coverage on `url-helpers.js`).

## Test plan

- [x] `npm test -w packages/spacecat-shared-utils` passes with 100% line/branch/function/statement coverage on `url-helpers.js`
- [x] Tests cover: absolute URLs, root path, trailing slash stripping, case normalization, query/fragment handling, invalid URL fallback, empty array edge case

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/spacecat-shared-utils/src/url-helpers.js [hunk 1]
```diff
@@ -512,6 +512,43 @@ function filterBySiteScope(urls, siteBaseUrl) {
   return urls.filter((url) => isWithinSiteScope(url, siteBaseUrl));
 }
 
+/**
+ * Extracts the pathname from a URL string, stripping trailing slashes on non-root paths.
+ * Falls back to the raw string when the URL is not parseable (e.g. invalid or relative).
+ *
```
Comment (MysticatBot) [inline packages/spacecat-shared-utils/src/url-helpers.js:519]: issue (blocking): These three functions are not added to the re-export block in `src/index.js`. Every other public function from this file (`canonicalizeUrl`, `filterBySiteScope`, etc.) is re-exported there. Without that addition, consumers importing from `@adobe/spacecat-shared-utils` cannot access the new helpers - only deep imports from the source file would work, which breaks the established consumption pattern.

**Fix:** Add `toPathname`, `hasSamePathname`, `allHaveSamePathname` to the `export { ... } from './url-helpers.js'` block in `src/index.js`.

File: packages/spacecat-shared-utils/src/url-helpers.js [hunk 2]
```diff
@@ -512,6 +512,43 @@ function filterBySiteScope(urls, siteBaseUrl) {
...
+ * @param {string} url
+ * @returns {string} pathname, or the original string on parse failure
+ */
+export function toPathname(url) {
+  try {
+    const { pathname } = new URL(url);
```
Comment (MysticatBot) [inline packages/spacecat-shared-utils/src/url-helpers.js:528]: issue (blocking): When `url` is `null`, `undefined`, or a non-string, `new URL(url)` throws (caught), but then `url.toLowerCase()` in the catch block also throws an unhandled `TypeError`. Neighboring functions in this file (`isWithinSiteScope`, `canonicalizeUrl`) guard against falsy/non-string inputs. Since `allHaveSamePathname` iterates a caller-supplied array, a single `null` element will crash the whole call.

**Fix:** Add an early guard:
```js
if (!url || typeof url !== 'string') return '';
```
Then add test cases for `toPathname(null)`, `toPathname(undefined)`, `hasSamePathname(null, url)`, and `allHaveSamePathname([null], url)`.

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @dhavkuma,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - two functional issues prevent the helpers from working correctly for package consumers.
**Complexity: LOW** - small utility addition, single package.
_Changes: Adds three URL pathname normalization/comparison helpers (`toPathname`, `hasSamePathname`, `allHaveSamePathname`) with tests (2 files)._

## Must fix before merge

1. [Important] Functions not re-exported from package entry point - `packages/spacecat-shared-utils/src/index.js` (details inline)
2. [Important] `toPathname` crashes on null/undefined input - `packages/spacecat-shared-utils/src/url-helpers.js:524` (details inline)

<details>
<summary>Non-blocking (3): minor issues and suggestions</summary>

- nit: Mixed export style - new functions use inline `export function` while the rest of the file uses a barrel `export {}` block at EOF - `packages/spacecat-shared-utils/src/url-helpers.js:518`
- suggestion: Add `if (!Array.isArray(urls)) return false;` guard on `allHaveSamePathname` to match the defensive style of sibling `filterBySiteScope` - `packages/spacecat-shared-utils/src/url-helpers.js:543`
- suggestion: `toPathname` JSDoc should mention the lowercasing behavior explicitly, since URL pathnames are technically case-sensitive per RFC 3986 and callers using `toPathname` directly (not just via `hasSamePathname`) may not expect it - `packages/spacecat-shared-utils/src/url-helpers.js:514`
</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 50s | <b>Cost:</b> $3.27 | <b>Commit:</b> <code>50001d69b7e847d51893904c67edd429885a64b9</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @dhavkuma,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - all prior review findings addressed, clean utility addition ready to merge.
**Complexity: LOW** - small utility addition, single package.
_Changes: Adds three URL pathname normalization/comparison helpers (`toPathname`, `hasSamePathname`, `allHaveSamePathname`) with comprehensive tests (4 files)._

<details>
<summary>Non-blocking (1): minor issues and suggestions</summary>

- nit: Trailing-slash normalization asymmetry in fallback branch - the happy path strips trailing slashes (`pathname.replace(/\/$/, '')`) but the catch branch returns `url.toLowerCase()` without stripping. This means `hasSamePathname('/path/', '/path')` is `false` while `hasSamePathname('https://x.com/path/', 'https://x.com/path')` is `true`. Documented behavior ("falls back to the raw string"), but potentially surprising if callers mix absolute and relative URLs - `packages/spacecat-shared-utils/src/url-helpers.js:525`
</details>

## Previously flagged, now resolved
- Functions now re-exported from package entry point (`src/index.js`)
- `toPathname` handles null/undefined gracefully (early guard added)
- `Array.isArray` guard added to `allHaveSamePathname`
- JSDoc updated to document lowercasing behavior
- Export structure consistent with existing patterns

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 39s | <b>Cost:</b> $3.07 | <b>Commit:</b> <code>ee6a1548fcecf5e70a579e1cb1a924fb851ff4e5</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>
